### PR TITLE
Replace usage of com.twitter.util.Time with java.time.Instant

### PR DESCRIPTION
**Pull request by jppierri**
(PD internal linked JIRA ticket: CORE-917)

Eliminate the dependency on Twitter Utils and use Java 8's DateTime utilities instead.

### CODE REVIEW

**Comment by jppierri** [comment]
Updated to use `java.time.Instant` rather than `java.time.LocalDateTime`.

File: common/src/main/scala/com/pagerduty/scheduler/model/TaskKey.scala [hunk 1]
```diff
@@ -40,18 +41,16 @@ case class TaskKey(
...
 
 object TaskKey {
-  // NOTE:
-  // Since com.twitter.util.Time is always stored internally as UTC, this format will always be UTC
-  // as well, even if the time was created with a different zone.
-  val ScheduledTimeFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
-  private[scheduler] val TimeFormat = new TimeFormat(ScheduledTimeFormat)
+  val ScheduledTimeFormat = "yyyy-MM-dd'T'HH:mm:ss[.SSS]'Z'"
+  private[scheduler] val TimeFormat =
+    DateTimeFormatter.ofPattern(ScheduledTimeFormat).withZone(ZoneOffset.UTC)
```
Comment (divtxt) [inline common/src/main/scala/com/pagerduty/scheduler/model/TaskKey.scala]: Any reason to not just use `DateTimeFormatter.ISO_INSTANT`? (which is [what Instant.parse uses](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#parse-java.lang.CharSequence-))

**Comment by divtxt** [comment]
LGTM.

**Comment by divtxt** [comment]
Fixed up merge conflict in `.travis.yml`

**Pull request merged**